emeval/input/phone_view.py

```diff
@@ -133,6 +133,11 @@ def filter_transitions(self, start_tt, end_tt, start_ti, end_ti,
     # We expect that transitions occur in pairs
     def transitions_to_ranges(transition_list, start_tt, end_tt, start_ti, end_ti,
             spec_end_ts):
+        # re-sort transitions into proper order -- START and STOP transitions must alternate
+        for i in range(0, len(transition_list), 2):
+            if "START" in transition_list[i]["transition"] and "START" in transition_list[i+1]["transition"] and "STOP" in transition_list[i+2]["transition"]:
```

Comment:
won't this fail at the end of the list?
when i is at the last element then it seems like `i+2` will go beyond the end of the list.

Reply:
I just tested this with

```
In [1]: x = [1,2,34]

In [5]: for i in range(0, len(x), 2):
   ...:     print(x[i], x[i+1], x[i+2])
   ...:
1 2 34
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-5-1a8c71f04d8e> in <module>
      1 for i in range(0, len(x), 2):
----> 2     print(x[i], x[i+1], x[i+2])
      3

IndexError: list index out of range
```

Reply:
Just in case it matters that the length is even

```
In [6]: x = [1,2,3, 4]

In [7]: for i in range(0, len(x), 2):
   ...:     print(x[i], x[i+1], x[i+2])
   ...:
   ...:
1 2 3
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-7-1049931a2401> in <module>
      1 for i in range(0, len(x), 2):
----> 2     print(x[i], x[i+1], x[i+2])
      3
      4

IndexError: list index out of range
```

Reply:
For the record, I manually applied this patch to my repo and it works. But I don't understand why. The prior code snippets suggest that it should fail

Reply:
@singhish  I know why this worked. It's because of "if" shortcircuiting - e.g. in 

```
In [19]: for i in range(0, len(x), 2):
    ...:     if x[i] > 10 and x[i+2] < 20:
    ...:         print(x[i], x[i+1], x[i+2])
    ...:
```

There is no error even though the `if` has an `[i+2]` index because the check before the `and` is always false. In our particular use case, we didn't encounter a `START`, `START` pattern at the end of the list, so the `[i+2]` check was never triggered. But if there were an enhanced dataset later that had that pattern, this check would have broken.

With the fix, it will not break.